From b2031fd4151cc70d5b74dcebba396be4e69d556f Mon Sep 17 00:00:00 2001
From: Ian Stakenvicius <axs@gentoo.org>
Date: Tue, 20 Nov 2012 07:35:00 -0800
Subject: [PATCH 07/24] immediate patch for 17ESR (minimal change, JS_VERSION
 bump)

FIXME: need commit message.
(Please also double check the author and subject.)

https://bugzilla.mozilla.org/show_bug.cgi?id=812265
---
 js/src/jsversion.h | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/js/src/jsversion.h b/js/src/jsversion.h
index 70e31a3..fbd2e3e 100644
--- a/js/src/jsversion.h
+++ b/js/src/jsversion.h
@@ -8,7 +8,7 @@
  * JS configuration macros.
  */
 #ifndef JS_VERSION
-#define JS_VERSION 185
+#define JS_VERSION 186
 #endif
 
 /*
@@ -136,6 +136,24 @@
 #define JS_HAS_GENERATOR_EXPRS  1       /* has (expr for (lhs in iterable)) */
 #define JS_HAS_EXPR_CLOSURES    1       /* has function (formals) listexpr */
 
+#elif JS_VERSION == 186
+
+#warning JS_VERSION is deprecated, please use MOZJS_MAJOR_VERSION et. al.
+#define JS_HAS_STR_HTML_HELPERS 1       /* has str.anchor, str.bold, etc. */
+#define JS_HAS_OBJ_PROTO_PROP   1       /* has o.__proto__ etc. */
+#define JS_HAS_OBJ_WATCHPOINT   1       /* has o.watch and o.unwatch */
+#define JS_HAS_TOSOURCE         1       /* has Object/Array toSource method */
+#define JS_HAS_CATCH_GUARD      1       /* has exception handling catch guard */
+#define JS_HAS_UNEVAL           1       /* has uneval() top-level function */
+#define JS_HAS_CONST            1       /* has JS2 const as alternative var */
+#define JS_HAS_FUN_EXPR_STMT    1       /* has function expression statement */
+#define JS_HAS_NO_SUCH_METHOD   1       /* has o.__noSuchMethod__ handler */
+#define JS_HAS_GENERATORS       1       /* has yield in generator function */
+#define JS_HAS_BLOCK_SCOPE      1       /* has block scope via let/arraycomp */
+#define JS_HAS_DESTRUCTURING    2       /* has [a,b] = ... or {p:a,q:b} = ... */
+#define JS_HAS_GENERATOR_EXPRS  1       /* has (expr for (lhs in iterable)) */
+#define JS_HAS_EXPR_CLOSURES    1       /* has function (formals) listexpr */
+
 #else
 
 #error "unknown JS_VERSION"
-- 
1.8.1.2

